test-runner/wasi_test_runner/test_reporters/__init__.py

```diff
@@ -0,0 +1,36 @@
+from abc import ABC
+from typing import List
+
+from ..test_case import TestCase
+from ..test_suite import TestSuite
+from ..runtime_adapter import RuntimeVersion
+
+
+class TestReporter(ABC):
+    def report_test(self, test: TestCase) -> None:
+        pass
+
+    def report_test_suite(self, test_suite: TestSuite) -> None:
+        pass
+
+    def finalize(self, version: RuntimeVersion) -> None:
+        pass
+
+
+class CompositeTestReporter(TestReporter):
```

Comment:
What's the benefit of this class above passing `List[TestReporter]` into `run_all_tests`, similar to the validators? Would prefer to avoid the indirection

Reply:
yeah, this class initially had a bit more responsibilities but as I was refactoring the code that got removed. I'll delete this class.